utils/src/test/java/software/amazon/awssdk/utils/cache/CachedSupplierTest.java

```diff
@@ -158,17 +195,58 @@ public void allCallsBeforeInitializationBlock() {
     }
 
     @Test
-    public void staleValueBlocksAllCalls() {
-        try (WaitingSupplier waitingSupplier = new WaitingSupplier(past(), future())) {
-            CachedSupplier<String> cachedSupplier = CachedSupplier.builder(waitingSupplier).build();
+    public void staleValueBlocksAllCalls() throws InterruptedException {
+        AdjustableClock clock = new AdjustableClock();
+        try (WaitingSupplier waitingSupplier = new WaitingSupplier(() -> now().plus(1, ChronoUnit.MINUTES), this::future)) {
+            CachedSupplier<String> cachedSupplier = CachedSupplier.builder(waitingSupplier)
+                                                                  .clock(clock)
+                                                                  .build();
 
             // Perform one successful "get".
             waitingSupplier.permits.release(1);
+            clock.time = now();
             waitFor(performAsyncGet(cachedSupplier));
 
             // Perform two "get"s that will attempt to refresh the value, and wait for them to get stuck.
-            performAsyncGets(cachedSupplier, 2);
+            clock.time = now().plus(61, ChronoUnit.SECONDS);
+            List<Future<?>> futures = performAsyncGets(cachedSupplier, 2);
+            waitingSupplier.waitForGetsToHaveStarted(3);
+            Thread.sleep(1_000);
+            assertThat(futures).allMatch(f -> !f.isDone());
+
+            // Release any "gets" that blocked and wait for them to finish.
+            waitingSupplier.permits.release(50);
+            waitForAsyncGetsToFinish();
+
+            // Make extra sure all 3 "gets" actually happened.
+            waitingSupplier.waitForGetsToHaveFinished(3);
+        }
+    }
+
+    @Test
+    public void staleValueBlocksAllCallsEvenWithStaleValuesAllowed() throws InterruptedException {
```

Comment:
This tests the entire point of this change: when we have static stability enabled, and we call get() on the supplier with a stale value, the get()s block.